eduNEXT/platform-plugin-turnitin
# feat: receive ora submission created event

### Description
This PR adds a receiver of the `ORA_SUBMISSION_CREATED` event. This receiver is responsible for running a celery task in the background that performs the following:

1. It obtains the ora submission information by its ID.
2. Each file (doc, docx, pdf) related to the submission is sent to Turnitin.
3. The response text related to the submission is converted to a .txt and sent to Turnitin.

Additionally:

1. The API was updated to receive the ORA submission ID, instead of the Turnitin submission ID. This is because a single ORA submission can be related to many Turnitin submissions.

### How to Test
1. Install this plugin in your environment.
2. Install edx-ora2 in your environment with the changes in [this PR](https://github.com/eduNEXT/edx-ora2/pull/11)
3. In a unit of the course create a new ora assessment, maybe **Staff Assessment Only**
4. In the settings of the component > **File Uploads Response = Required** and in **File Types = pdf,doc,docx**
5. In the LMS send a submission with some files.
6. Consumes this endpoint:

    ```
    {lms_domain}/platform-plugin-turnitin/{course_id}/api/v1/submission/{ora_submission_id}/
    ```
7. You should see a list with Turnitin submission information for all files related to ora submission.
